plugins/catalog-backend-module-gitlab/src/providers/GitlabDiscoveryEntityProvider.ts

```diff
@@ -185,22 +192,19 @@ export class GitlabDiscoveryEntityProvider implements EntityProvider {
         continue;
       }
 
-      const project_branch =
-        this.config.branch ??
-        project.default_branch ??
-        this.config.fallbackBranch;
-
       const projectHasFile: boolean = await client.hasFile(
         project.path_with_namespace ?? '',
-        project_branch,
+        this.branchSelector(project.default_branch),
         this.config.catalogFile,
       );
       if (projectHasFile) {
         res.matches.push(project);
       }
     }
 
-    const locations = res.matches.map(p => this.createLocationSpec(p));
+    const locations = res.matches
+      .map(project => this.createLocationUrl(project))
+      .map(targetUrl => this.createLocationFn(targetUrl));
```

Comment:
Here we have it iterating through our locations list twice, but I think the way it was can work by just calling `this. createLocationFn` in `createLocationSpec`

Reply:
So you mean like this?
```
const locations = res.matches
      .map(project => this.createLocationFn(this.createLocationUrl(project)));
```

Reply:
Like the way it was before
```suggestion
      const locations = res.matches.map(p => this.createLocationSpec(p));
```

But just

```
private createLocationSpec(project: GitLabProject): LocationSpec {
    const project_branch =
      this.config.branch ??
      project.default_branch ??
      this.config.fallbackBranch;
    
    return this.createLocationFn();
  }
```